src/display-x11.cc

```diff
@@ -675,33 +708,30 @@ EV_HANDLER(damage) {
                     x11_stuff.part);
   return true;
 }
-#else
-NOOP_EV_HANDLER(damage)
 #endif /* BUILD_XDAMAGE */
 
 /// Handles all events conky can receive.
 ///
 /// @return true if event should move input focus to conky
 bool process_event(conky::display_output_x11 *surface, Display *display,
                    XEvent ev, bool *consumed, void **cookie) {
-#define HANDLE_EV(handler)                                                \
-  if (_conky_ev_handle_##handler(surface, display, ev, consumed, cookie)) \
-  return true
+#define HANDLE_EV(event)                                                   \
+  if (handle_event<x_event_handler::event>(surface, display, ev, consumed, \
+                                           cookie)) {                      \
+    return true;                                                           \
+  }
```

Comment:
Can I keep a macro here?

It's right next to invocation, won't be called anywhere else, and quarters the code below. I would use a constexpr for loop to loop over all `x_event_handler` values, but those don't exist in C++.